Add support for y-prosemirror history

jevakallio opened the pull request:
Hey @smoores-dev, thank you for your work on this package! Great work!

We're just starting to integrate it into our application, and we needed to add support for `y-prosemirror`'s custom undo/redo mechanism. Since prosemirror is often used in collaborative scenarios, this might be useful for other users of the library as well.

smoores-dev commented:
Looks good to me @jevakallio! Could you fix that typescript-eslint issue, and also run `yarn version patch` and check in the resulting version file? Thanks!

jevakallio commented:
Oops, missed your response! I've addressed the lint error, plus also fixed a related issue -- if one client has enabled syncing, other clients' updates were transformed into suggestions. In the y-prosemirror case, this was the entire document, given that the plugin replaces the entire document content in its entirety.
